Sources/Gui/Main.cpp

```diff
@@ -57,13 +57,23 @@
 #endif
 
 
+#ifdef _MSC_VER>=1900	// Visual Studio 2015 or higher
+extern "C" { FILE __iob_func[3] = { *stdin,*stdout,*stderr }; }
```

Comment:
It did work for me (without any warnings or anything).
It acts as an array of pointers to stdin, stdout and stderr. `FILE` is need for some reason, instead of just a pointer, but it works.
Apparently there's still some problems with resource loading (particularly with Icon.inc and SplashImage.inc because of the way they're loaded at compile time), but this PR is a start to VS2015 compatibility, it takes care of the most important problems.